ui/app/AppLayouts/Communities/popups/TokenMasterActionPopup.qml

```diff
@@ -0,0 +1,178 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+import QtQml.Models 2.15
+
+import StatusQ.Core 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Popups.Dialog 0.1
+import StatusQ.Core.Theme 0.1
+
+import AppLayouts.Communities.panels 1.0
+
+import utils 1.0
+
+
+StatusDialog {
+    id: root
+
+    enum ActionType {
+        RemotelyDestruct, Kick, Ban
+    }
+
+    title: {
+        if (actionType === TokenMasterActionPopup.ActionType.Ban)
+            return qsTr("Ban %1").arg(userName)
+        if (actionType === TokenMasterActionPopup.ActionType.Kick)
+            return qsTr("Kick %1").arg(userName)
+
+        qsTr("Remotely destruct TokenMaster token")
+    }
+
+    implicitWidth: 600
+
+    property int actionType: TokenMasterActionPopup.ActionType.RemotelyDestruct
+
+    property string userName
+    property string networkName
+
+    property string feeText
+    property string feeErrorText
+    property bool isFeeLoading
+    property var accountsModel
+
+    readonly property string feeLabel: qsTr("Remotely destruct 1 TokenMaster token on %1").arg(
+                                           root.networkName)
+
+    signal remotelyDestructClicked
+    signal kickClicked
+    signal banClicked
+
+    ColumnLayout {
+        anchors.fill: parent
+        spacing: 20
+
+        StatusBaseText {
+            Layout.fillWidth: true
+            text: {
+                if (root.actionType === TokenMasterActionPopup.ActionType.RemotelyDestruct)
+                    return qsTr('Continuing will destroy the TokenMaster token held by <span style="font-weight:600;">%1</span> and revoke the permissions they have by virtue of holding this token.').arg(root.userName)
+
+                if (root.actionType === TokenMasterActionPopup.ActionType.Kick)
+                    return qsTr('Are you sure you kick <span style="font-weight:600;">%1</span> from Doodles? <span style="font-weight:600;">%1</span> is a TokenMaster hodler. In order to kick them you must also remotely destruct their TokenMaster token to revoke the permissions they have by virtue of holding this token.').arg(root.userName)
```

Comment:
Doodles here! :)

Reply:
Omg :man_facepalming: Thanks for point that!